samples/MASMessagingSample/src/main/java/com/ca/mas/masmessagingsample/activity/SearchGroupActivity.java

```diff
@@ -0,0 +1,148 @@
+                        Snackbar.make(getWindow().getDecorView(), "Please enter a valid group name", Snackbar.LENGTH_SHORT).show();
+                        return false;
+                    }
+                }
+                MASGroup.newInstance().getGroupMetaData(new MASCallback<GroupAttributes>() {
+                    @Override
+                    public void onSuccess(GroupAttributes result) {
+                        searchForGroup(result);
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        Log.e(TAG, "Failed to get group attributes: " + e);
+                        Snackbar.make(getWindow().getDecorView(), "Unable to fetch group attributes:" + e.getMessage(), Snackbar.LENGTH_SHORT).show();
+                    }
+                });
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    private void searchForGroup(GroupAttributes attributes) {
+        final MASFilteredRequest request = new MASFilteredRequest(attributes.getAttributes(),
+                IdentityConsts.KEY_GROUP_ATTRIBUTES);
+        request.isEqualTo(IdentityConsts.KEY_GROUPS_BY_NAME, mSearchView.getText().toString().trim());
+
+        MASGroup.newInstance().getGroupsByFilter(request, new MASCallback<List<MASGroup>>() {
+
+            @Override
+            public Handler getHandler() {
+                return new Handler(Looper.getMainLooper());
+            }
+
+            @Override
+            public void onSuccess(List<MASGroup> result) {
+                if (result == null || result.isEmpty()) {
+                    Snackbar.make(getWindow().getDecorView(), "No groups found with this name", Snackbar.LENGTH_SHORT).show();
+                    return;
+                }
+
+                LinearLayoutManager llm = new LinearLayoutManager(mContext);
+                llm.setOrientation(LinearLayoutManager.VERTICAL);
+                mRecyclerView.setLayoutManager(llm);
+
+                mResultList = getFilteredList(result);
+                if (mResultList == null || mResultList.isEmpty()) {
+                    Snackbar.make(getWindow().getDecorView(), "You are not member of this group", Snackbar.LENGTH_SHORT).show();
+                    return;
+                }
+                DataManager.INSTANCE.addGroups(mResultList);
+                GroupRecyclerAdapter adapter = new GroupRecyclerAdapter(mResultList, SearchGroupActivity.class.getName());
+                mRecyclerView.setAdapter(adapter);
+                mRecyclerView.addItemDecoration(new DividerDecoration(mContext));
+            }
+
+            @Override
+            public void onError(Throwable e) {
+                Log.e(TAG, "Failed to search for groups: " + e);
+                Snackbar.make(getWindow().getDecorView(), "Failed to search for groups:" + e.getMessage(), Snackbar.LENGTH_SHORT).show();
```

Comment:
Add a space after "Failed to search for groups:" to become ""Failed to search for groups: "